aptos-move/framework/aptos-stdlib/sources/cryptography/ed25519.move

```diff
@@ -166,6 +170,44 @@ module aptos_std::ed25519 {
         std::hash::sha3_256(pk_bytes)
     }
 
+    #[test_only]
+    public fun generate_keys(): (SecretKey, ValidatedPublicKey) {
+        let (sk_bytes, pk_bytes) = generate_keys_internal();
+        let sk = SecretKey {
+            bytes: sk_bytes
+        };
+        let pk = ValidatedPublicKey {
+            bytes: pk_bytes
+        };
+        (sk,pk)
+    }
+
+    #[test_only]
+    public fun sign_arbitrary_bytes(sk: &SecretKey, msg: vector<u8>): Signature {
+        Signature {
+            bytes: sign_internal(sk.bytes, msg)
+        }
+    }
+
+    #[test_only]
+    public fun sign_struct<T:drop>(sk: &SecretKey, data: T): Signature {
+        let encoded = new_signed_message(data);
+        Signature {
+            bytes: sign_internal(sk.bytes, bcs::to_bytes(&encoded))
+        }
+    }
+
+    #[test]
+    fun test_gen_sign_verify_combo() {
+        let (sk, vpk) = generate_keys();
+        let pk = public_key_into_unvalidated(vpk);
+        let msg: vector<u8> = x"0123456789abcdef";
+        let sig1 = sign_arbitrary_bytes(&sk, msg);
+        assert!(signature_verify_strict(&sig1, &pk, msg), std::error::invalid_state(1));
+        let sig2 = sign_struct(&sk, copy pk);
+        assert!(signature_verify_strict_t(&sig2, &pk, copy pk), std::error::invalid_state(2));
```

Comment:
Let's do a clearer example here where we sign an actual struct. Signing `pk` and verifying with `pk` will not be very education for people reading this code.

Instead, have a `#[test_only]` struct `TestMessage` declared above with a `bytes` field and maybe a `number` field and sign an instantiation of that instead.